components/director/run.sh

```diff
@@ -89,10 +92,17 @@ function cleanup() {
     else
         echo -e "${GREEN}Skipping Postgres container cleanup${NC}"
     fi
+
+    echo -e "${GREEN}Destroying k3d cluster..."
+    k3d cluster delete k3d-cluster
 }
 
 trap cleanup EXIT
 
+echo -e "${GREEN}Creating k3d cluster..."
+curl -s https://raw.githubusercontent.com/rancher/k3d/main/install.sh | TAG=v5.2.0 bash
+k3d cluster create k3d-cluster --api-port 6550 --servers 1 --port 443:443@loadbalancer --image rancher/k3s:v1.22.4-k3s1 --kubeconfig-update-default --wait
+
```

Comment:
I personally don't like this approach as it will make the run.sh significantly slower and heavier while the benefit is nearly zero (no one is actually using local run of director for testing externally issued certs).

If we would like to leave it like this at least we need a way to make it optional, otherwise, the whole purpose of the script is gone as it will become more and more like minikube installation with the director only.

Reply:
That was my concern as well. But we've discussed it on our daily and decide to check how effort it will be and how much time it will needs to start. This cluster creation takes around 15 seconds which I think it's acceptable so far? But we can discuss it again?

Reply:
We've discussed it offline and agreed to use this approach because the cluster creation time is minimal and we can use it for further improvements/testing/debugging